crates/client/data-availability/src/lib.rs

```diff
@@ -206,11 +206,9 @@ pub async fn update_state<B: BlockT, H: HasherT>(
     block_da_data: BlockDAData,
 ) -> Result<(), anyhow::Error> {
     let block_hash = block_da_data.block_hash;
+    let calldata = block_data_to_calldata(block_da_data);
     // store the state diff
-    madara_backend
-        .da()
-        .store_state_diff(&block_hash, state_diff_to_calldata(block_da_data))
-        .map_err(|e| anyhow!("{e}"))?;
+    madara_backend.da().store_state_diff(&block_hash, calldata.clone()).map_err(|e| anyhow!("{e}"))?;
```

Comment:
So we are still storing the calldata. With a method named `store_state_diff`

Reply:
I guess we didn't decide anything clearly. But what I had in mind was.

Store stuffs in the db scale-encoded, read them, convert them to calldata only when you actually need calldata.

@apoorvsadana

Reply:
I personnaly don't see why it's not fine to store the state diff encoded as calldata, it's the best intermediate representation we have and consumers can just decode it into whatever struct they need.

Reply:
But the SCALE encoding is already an intermediate representation. Why do we need to create another one i.e. calldata, if whenever it is read, we have to decode it anyways? Like, if there was a use case where we were reading the stored data and encoding it into call data, I think it would make more sense. However, right now, we encode it, store it and then later fetch it and decode it. My point is, the intermediate representation isn't solving much and it makes it difficult to edit later on.

Reply:
the calldata repr is used to push on ethereum. 
A similar logic would mean store everything else as JSON because we are going to convert it to JSON in the RPC return type. 
Doesn't make sense. We should store it as the most polyvalent repr as much as possible, which given our diversity of usage, is scale encode. As it is easily decoded to the rust types

Reply:
> the calldata repr is used to push on ethereum.
Yes, but the function where we do the encoding is also the function where we push it. So, its in memory already at that point when we are pushing the encoded state to Ethereum. So the what we store is being used for other places like for the RPC. Over there, it makes more sense to read directly a Rust type as its easily readable and changeable later on. In the current form, if I want to change this I need to change the decode and encode functions as well.

Reply:
> the calldata repr is used to push on ethereum.

Yes, but the function where we do the encoding is also the function where we push it. So, its in memory already at that point when we are pushing the encoded state to Ethereum. So the what we store is being used for other places like for the RPC. Over there, it makes more sense to read directly a Rust type as its easily readable and changeable later on. In the current form, if I want to change this I need to change the decode and encode functions as well.

Reply:
So I agree that we should store in a polyvalent form and that SCALE already solves that for us. I am saying that if it was the case that whenever we were reading state diffs we were encoding them to make them usable, then it would make sense to just store the encoded form. For example, if there was one worker storing the diff and then there were multiple workers reading it and encoding it, then I would vote for encoding it and storing it. In the current case, I feel storing the rust type directly solves the problem better.

Reply:
Ended up storing `ThinStateDiff` in the DA db, waiting for https://github.com/keep-starknet-strange/starknet-api/pull/15 to be merged

This way we can get rid of the `decode_v011` function but we now have a `to_rpc_state_diff` conversion util